controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -115,12 +115,34 @@ else if (retryBecauseNewFailure(application, report.jobType()))
         });
     }
 
-    /** Returns whether all production zones listed in deployment spec last were successful on the currently deploying change. */
+    /** Returns whether all production zones listed in deployment spec has this change (or a newer version, if upgrade) */
     private boolean deploymentComplete(LockedApplication application) {
         if ( ! application.deploying().isPresent()) return true;
```

Comment:
The comment here was hidden because of a new commit, so I'll answer again:

No, it returns `true`.

Reply:
Copying my answer:
No but perhaps you mean if it was cancelled and then later we check for another Change which is an upgrade to an earlier version? Then it will return true but then it is also true that the upgrade is complete ...